### DISCUSSION TITLE
4/17 (월) 제이미 - Configuration과 TestConfiguration 빈 등록 시 충돌 해결 방법 #3

### DISCUSSION THREAD

**Discussion by JJ503** [woowacourse-study/2023-presentation-study#6] [Q&A]
[자료](https://treasure-fountain-650.notion.site/Configuration-TestConfiguration-35cf973bea2542e48ad96ddbec7fed1d)

**Comment by jyeost** [comment]
3번 해결법에서 메서드 명이 동일하면 오류가 나나요?

**Reply by JJ503** [reply]
`@Component`는 클래스의 인스턴스를, `@Bean`은 객체를 반환하기 때문에 충돌될 문제는 없을 것입니다!
다만, `@TestConfiguration`을 통해 등록된 빈을 사용에 있어 우선시하기에 `@Component`로 등록된 빈이 아닌 `@TestConfiguration`을 통해 등록된 빈을 사용하게 될 것입니다.

**Comment by yeonkkk** [comment]
3가지 해결 방법을 제시해주셨는데, 제이미가 생각했을 때 어떤 방법이 가장 이상적이라고 생각하시나요?

**Reply by JJ503** [reply]
발표에서 예시에서는 3번째 `@Component`를 사용한 방법이 가장 적절하다고 생각합니다.
예시의 코드는 외부 라이브러리 코드가 아닌 직접 작성한 코드이기에 빈을 등록하는 코드를 작성할 필요 없이 `@Component`를 사용해 자동으로 빈을 등록하는 것이 더 깔끔하다고 생각합니다.

**Reply by yeonkkk** [reply]
답변 감사합니다 제이미!!👍
